src/main/java/com/commercetools/sync/producttypes/utils/ProductTypeUpdateAttributeDefinitionActionUtils.java

```diff
@@ -0,0 +1,279 @@
+            .map(AttributeDefinitionDraft::getName)
+            .collect(toList());
+
+        final List<String> existingNames = oldAttributeDefinitions
+            .stream()
+            .map(AttributeDefinition::getName)
+            .filter(oldName -> newNames.contains(oldName))
+            .collect(toList());
+
+        final List<String> notExistingNames = newNames
+            .stream()
+            .filter(newName -> !existingNames.contains(newName))
+            .collect(toList());
+
+        final List<AttributeDefinition> newAttributeDefinitionsOrder = newAttributeDefinitionDrafts
+            .stream()
+            .map(AttributeDefinitionCustomBuilder::of)
+            .collect(toList());
+
+        final List<String> allNames = Stream.concat(existingNames.stream(), notExistingNames.stream())
+            .collect(toList());
+
+        return buildUpdateAction(
+            allNames,
+            newNames,
+            () -> productTypeAttributeDefinitionActionFactory
+                .buildChangeAttributeDefinitionOrderAction(newAttributeDefinitionsOrder)
+        );
+
+
+    }
+
+    /**
+     * Checks if there are any new attribute definition drafts which are not existing in the
+     * {@code oldAttributeDefinitionNameMap}. If there are, then "add" attribute definition update actions are built
+     * using the instance of {@link ProductTypeAttributeActionFactory} supplied to add the
+     * missing attribute definition. Otherwise, if there are no new attribute definitions, then an empty list
+     * is returned.
+     *
+     * @param newAttributeDefinitionDrafts                  the list of new {@link AttributeDefinitionDraft}s.
+     * @param oldAttributeDefinitionNameMap                 a map of names to AttributeDefinition of the old list
+     *                                                      of attribute definition.
+     * @param productTypeAttributeDefinitionActionFactory   factory responsible for building attribute definition update
+     *                                                      actions.
+     * @return a list of attribute definition update actions if there are new attribute definition that should be added.
+     *         Otherwise, if the attribute definitions are identical, an empty optional is returned.
+     */
+    @Nonnull
+    private static List<UpdateAction<ProductType>> buildAddAttributeDefinitionUpdateActions(
+        @Nonnull final List<AttributeDefinitionDraft> newAttributeDefinitionDrafts,
+        @Nonnull final Map<String, AttributeDefinition> oldAttributeDefinitionNameMap,
+        @Nonnull final ProductTypeAttributeActionFactory productTypeAttributeDefinitionActionFactory) {
+
+        // Bug in the commercetools JVM SDK. AddAttributeDefinition should expect an AttributeDefinitionDraft rather
+        // than AttributeDefinition. That's why we use AttributeDefinitionCustomBuilder
+        // It will be fixed in https://github.com/commercetools/commercetools-jvm-sdk/issues/1786
```

Comment:
Ah now I noticed you added a comment about that here :) A `TODO` would be useful too, since the IDE can then help us about it.